python/lsst/summit/utils/astrometry/utils.py

```diff
@@ -0,0 +1,288 @@
+# This file is part of summit_utils.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+
+import astropy.units as u
+from astropy.time import Time
+from astropy.coordinates import AltAz, EarthLocation, SkyCoord
+
+from lsst.afw.geom import SkyWcs
+from lsst.daf.base import PropertySet
+from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask, CharacterizeImageConfig
+
+__all__ = [
+    'chuckHeaderToWcs',
+    'getAverageRaFromHeader',
+    'getAverageDecFromHeader',
+    'getAverageAzFromHeader',
+    'getAverageElFromHeader',
+    'genericCameraHeaderToWcs',
+    'getIcrsAtZenith',
+    'headerToWcs',
+    'runCharactierizeImage',
+    'filterSourceCatOnBrightest',
+]
+
+
+def chuckHeaderToWcs(exp, nominalRa=None, nominalDec=None):
+    """Given an exposure taken by Chuck at his house, construct a wcs
+    with the ra/dec set to zenith unless a better guess is supplied.
+
+    Automatically sets the platescale depending on the lens.
+
+    Parameters
+    ----------
+    exp : `lsst.afw.image.Exposure`
+        The exposure to construct the wcs for.
+    nominalRa : `float`, optional
+        The guess for the ra.
+    nominalDec : `float`, optional
+        The guess for the Dec.
+
+    Returns
+    -------
+    wcs : `lsst.afw.geom.SkyWcs`
+        The constructed wcs.
+    """
+    header = exp.getMetadata().toDict()
+
+    # set the plate scale depending on the lens and put into CD matrix
+    # plate scale info from:
+    # https://confluence.lsstcorp.org/pages/viewpage.action?pageId=191987725
+    lens = header['INSTLEN']
+    if '135mm' in lens:
+        arcSecPerPix = 8.64
+    elif '375mm' in lens:
+        arcSecPerPix = 3.11
+    elif '750mm' in lens:
+        arcSecPerPix = 1.56
+    else:
+        raise ValueError(f'Unrecognised lens: {lens}')
+
+    header['CD1_1'] = arcSecPerPix / 3600
```

Comment:
Pretend you hear a distance voice saying "use astropy units!" to which you and I reply "you can't make me!".

Reply:
Haha! That particular function was really only useful when the StarTrackers were in Chuck's garden anyway 🙂